src/main/java/com/leveluptasks/service/GroupService.java

```diff
@@ -37,50 +41,40 @@ public Groupe getGroupById(Long id) {
         return groupRepository.findById(id).orElse(null);
     }
     
-      
-    public String saveGroup( GroupCreationDTO groupCreationDTO) {
+    public String saveGroup(GroupCreationDTO groupCreationDTO) {
         Groupe groupe = new Groupe();
         groupe.setName(groupCreationDTO.getGroupName());
-        Set<UserHasGroup> userHasGroups = new HashSet<>();
-        List<String> userNames = groupCreationDTO.getUserNames();
-        String result = "KO";
-        if (userNames != null) {
-            for (String userName : userNames) {
-                User user = userRepository.findByFirstname(userName);
-                if (user != null) {
-                    UserHasGroup userHasGroup = new UserHasGroup();
-                    userHasGroup.setUser(user);
-                    userHasGroup.setGroupe(groupe);
-                    userHasGroups.add(userHasGroup);
-                    groupe.setUserHasGroups(userHasGroups);
-                    result = "OK";
-                }else {
-                    result = "KO";
-                    throw new IllegalArgumentException("Utilisateur avec le prénom '" + userName + "' non trouvé.");
-                }
+
+        String userEmail = groupCreationDTO.getUserEmail();
+        if (userEmail != null && !userEmail.isEmpty()) {
+            Optional<User> userOptional = userRepository.findByEmail(userEmail);
+            if (userOptional.isPresent()) {
+                User user = userOptional.get();
+                UserHasGroup userHasGroup = new UserHasGroup();
+                userHasGroup.setUser(user);
+                userHasGroup.setGroupe(groupe);
+
+                Set<UserHasGroup> userHasGroups = new HashSet<>();
+                userHasGroups.add(userHasGroup);
+
+                groupe.setUserHasGroups(userHasGroups);
+            } else {
+                throw new IllegalArgumentException("User not found with email: " + userEmail);
             }
         }
-        groupRepository.save(groupe);
-        return result;
 
-      
+        groupRepository.save(groupe);
+        return "OK";
```

Comment:
better return the group itself if you want to display it, or at least a response entity. Beter than a simple String